framework/encode/capture_manager.cpp

```diff
@@ -700,6 +712,26 @@ void CaptureManager::WriteDisplayMessageCmd(const char* message)
     }
 }
 
+void CaptureManager::WriteAnnotation(const format::AnnotationType type, const char* label, const char* data)
+{
+    if ((capture_mode_ & kModeWrite) == kModeWrite)
+    {
+        const auto label_length = decltype(format::AnnotationHeader::label_length)(util::platform::StringLength(label));
```

Comment:
Sorry I didn't catch this previously; should this also be a `static_cast` and use `GFXRECON_CHECK_CONVERSION_DATA_LOSS`?

Reply:
I got rid of a lot of this mess and made it much simpler. Hopefully I didn't also mess up the 32 bit win build again.